floater/test/test_generators.py

```diff
@@ -186,6 +187,49 @@ def test_pickling_with_land(fs_with_land, tmpdir):
                 assert np.all(fs.__dict__[key][sub_key] == fs_from_pickle.__dict__[key][sub_key])
 
 
+def test_npart_to_2D_array():
+    lon = np.linspace(0, 8, 9, dtype=np.float32)
+    lat = np.linspace(-4, 4, 9, dtype=np.float32)
+    land_mask = np.zeros(81)
+    for i in range(81):
+        if (i%9==0) or (i%9==1):
+            land_mask[i] = 0.
+        else:
+            land_mask[i] = 1.
+    land_mask = land_mask==1.
+    land_mask.shape = (len(lat), len(lon))
+    model_grid = {'lon': lon, 'lat': lat, 'land_mask': land_mask}
+    fs = gen.FloatSet(xlim=(0, 9), ylim=(-4, 5), dx=1, dy=1, model_grid=model_grid)
+    fs.get_rectmesh()
+    var_list = ['test_01', 'test_02', 'test_03']
+    data_vars = {}
+    for var in var_list:
+        values = np.random.random(69)
+        data_vars.update({var: (['npart'], values)})
+    npart = np.linspace(1, 69, 69, dtype=np.int32)
+    coords = {'npart': (['npart'], npart)}
+    ds1d = xr.Dataset(data_vars=data_vars, coords=coords)
+    da1d = ds1d['test_01']
+    # method test
+    da2d = fs.npart_to_2D_array(da1d)
+    ds2d = fs.npart_to_2D_array(ds1d)
+    # shape test
+    assert da2d.to_array().values.shape == (1, fs.Ny, fs.Nx)
+    assert ds2d.to_array().values.shape == (3, fs.Ny, fs.Nx)
+    # coordinates test
+    assert list(da2d.lon.values) == list(fs.x)
+    assert list(da2d.lat.values) == list(fs.y)
+    assert list(ds2d.lon.values) == list(fs.x)
+    assert list(ds2d.lat.values) == list(fs.y)
+    # mask test
+    mask1d = list(fs.ocean_bools)
+    mask2d_da = list((np.isnan(da2d.to_array().values[0])==False).ravel())
+    assert mask2d_da == mask1d
+    for i in range(3):
+        mask2d_ds = list((np.isnan(ds2d.to_array().values[i])==False).ravel())
+        assert mask2d_ds == mask1d
+
+
```

Comment:
What happens if there is no land mask at (i.e. no `model_grid` keyword is passed to `FloatSet`)? Can we also test for this?